initialization/initialization_test.go

```diff
@@ -498,32 +515,37 @@ func TestInitialize_NumUnits_MultipleFiles(t *testing.T) {
 		eg.Wait()
 	}
 
-	prevNumUnits := opts.NumUnits
-
-	// Increase `opts.NumUnits` while `opts.NumFiles` > 1.
-	opts.NumUnits = prevNumUnits + 1
-	_, err = NewInitializer(
+	// Decrease `opts.NumUnits`.
+	newOpts := opts
+	newOpts.NumUnits = opts.NumUnits - 1
+	init, err = NewInitializer(
 		WithNodeId(nodeId),
 		WithCommitmentAtxId(commitmentAtxId),
 		WithConfig(cfg),
-		WithInitOpts(opts),
+		WithInitOpts(newOpts),
 		WithLogger(testLogger{t: t}),
 	)
-	var cfgMissErr shared.ConfigMismatchError
-	r.ErrorAs(err, &cfgMissErr)
-	r.Equal("NumUnits", cfgMissErr.Param)
+	r.NoError(err)
 
-	// Decrease `opts.NumUnits` while `opts.NumFiles` > 1.
-	opts.NumUnits = prevNumUnits - 1
-	_, err = NewInitializer(
-		WithNodeId(nodeId),
-		WithCommitmentAtxId(commitmentAtxId),
-		WithConfig(cfg),
-		WithInitOpts(opts),
-		WithLogger(testLogger{t: t}),
-	)
-	r.ErrorAs(err, &cfgMissErr)
-	r.Equal("NumUnits", cfgMissErr.Param)
+	{
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		var eg errgroup.Group
+		eg.Go(assertNumLabelsWritten(ctx, t, init))
+
+		numFiles, err := init.diskState.NumFiles()
+		r.NoError(err)
+		r.Equal(int(opts.NumUnits), numFiles)
```

Comment:
This assertion looks logically incorrect. Why should the number of files equal the number of PoST units?

```suggestion
		layout := config.DeriveFilesLayout(cfg, opts)
		r.Equal(int(layout.NumFiles), numFiles)
```

Reply:
It works because of the actual `cfg` & `opts` values, but I agree that it's confusing. Fixed.